step/step.go

```diff
@@ -197,7 +200,42 @@ func (s XcodeTestConfigParser) ProcessConfig() (Config, error) {
 		return Config{}, fmt.Errorf("`-xcconfig` option found in 'Additional options for the xcodebuild command' (xcodebuild_options), please clear 'Build settings (xcconfig)' (`xcconfig_content`) input as only one can be set")
 	}
 
-	return s.utils.CreateConfig(input, projectPath, sim, additionalOptions, additionalLogFormatterOptions), nil
+	skipTesting, err := s.processQuarantinedTests(input.QuarantinedTests)
+	if err != nil {
+		return Config{}, fmt.Errorf("failed to process quarentined tests: %w", err)
+	}
+
+	return s.utils.CreateConfig(input, projectPath, sim, additionalOptions, additionalLogFormatterOptions, skipTesting), nil
+}
+
+/*
+processQuarantinedTests converts the Bitrise quarantined tests JSON input ($BITRISE_QUARANTINED_TESTS_JSON)
+to test identifiers for the `-skip-testing` xcodebuild option. The test identifier format is: <TestTarget>/<TestClass>/<TestMethod>.
+*/
+func (s XcodeTestConfigParser) processQuarantinedTests(quarantinedTestsInput string) ([]string, error) {
+	if quarantinedTestsInput == "" {
+		return nil, nil
+	}
+
+	quarantinedTests, err := testquarantine.ParseQuarantinedTests(quarantinedTestsInput)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse quarantined tests input: %w", err)
+	}
+
+	var skippedTests []string
+	for _, qt := range quarantinedTests {
+		if len(qt.TestSuiteName) == 0 || qt.TestSuiteName[0] == "" || qt.ClassName == "" || qt.TestCaseName == "" {
+			continue
+		}
+
+		testTarget := qt.TestSuiteName[0]
```

Comment:
I made this comment already on an another PR so I just leave it here as a placeholder.

Why is this an array if we only take the first item out of it?

Reply:
I think this is based on how we transform the XCode results into our internal test data (which also explains why suite name, classname and test case name are all mandatory, which is not generally true). It's an array because it could be hierarchical, but that never happens with this step. Right, @godrei ?

Reply:
Yes, this the intermediate test data, that we create from an original xcode test result: [TestSuite](https://github.com/bitrise-steplib/steps-deploy-to-bitrise-io/blob/2d1faa94e4fe765a34d87c3e48367e747b68caf3/test/converters/xcresult3/model3/data.go#L19) this test suit, doesn't have subtest suites, just test cases.
(And this also applies to the [legacy Xcode test result parsing](https://github.com/bitrise-steplib/steps-deploy-to-bitrise-io/blob/f91469ccdac2b6d3eaa03ccbc3a5262370c9771a/test/converters/xcresult3/converter.go#L410-L417))